src/editor/css_sort.rs

```diff
@@ -0,0 +1,103 @@
+use scraper::Selector;
+use serde::Deserialize;
+use url::Url;
+
+use super::Editor;
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct CssSort {
+    #[serde(deserialize_with = "super::deserialize_selector")]
+    pub selector: Selector,
+
+    #[serde(default)]
+    pub reverse: bool,
+
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub sort_by: Vec<Editor>,
+}
+
+impl CssSort {
+    pub fn apply(&self, url: &Url, html: &str) -> anyhow::Result<String> {
+        let parsed_html = scraper::Html::parse_document(html);
+        let mut selected = parsed_html.select(&self.selector).collect::<Vec<_>>();
+
+        if selected.is_empty() {
+            anyhow::bail!("selected nothing");
+        }
+
+        selected.sort_by_cached_key(|item| {
+            let mut content = super::Content {
+                extension: Some("html"),
+                text: item.html(),
+            };
+            for editor in &self.sort_by {
+                if let Ok(inner) = editor.apply(url, content) {
+                    content = inner;
+                } else {
+                    return String::new();
+                }
+            }
+            content.text
+        });
+
+        if self.reverse {
+            selected.reverse();
+        }
+
+        Ok(selected
+            .iter()
+            .map(scraper::ElementRef::html)
+            .collect::<Vec<_>>()
+            .join("\n"))
+    }
+}
+
+#[test]
+fn simple_example() {
+    let url = Url::parse("https://edjopato.de/").unwrap();
+    let input = "<html><head></head><body><p>A</p><p>C</p><p>B</p></body></html>";
+    let expected = "<p>A</p>
+<p>B</p>
+<p>C</p>";
+    let example = CssSort {
+        selector: Selector::parse("p").unwrap(),
+        sort_by: Vec::new(),
+        reverse: false,
+    };
+    let html = example.apply(&url, input).unwrap();
+    assert_eq!(html, expected);
+}
+
+#[test]
+fn simple_example_reverse() {
+    let url = Url::parse("https://edjopato.de/").unwrap();
+    let input = "<html><head></head><body><p>A</p><p>C</p><p>B</p></body></html>";
+    let expected = "<p>C</p>
+<p>B</p>
+<p>A</p>";
+    let example = CssSort {
+        selector: Selector::parse("p").unwrap(),
+        sort_by: Vec::new(),
+        reverse: true,
+    };
+    let html = example.apply(&url, input).unwrap();
+    assert_eq!(html, expected);
+}
+
+#[test]
+fn sort_by_example() {
```

Comment:
```suggestion
fn sort_by_html_child_using_an_additional_editor() {
```

Reply:
This is kind of obvious. `sort_by` works by editors.

Reply:
No it is not. And "works by editors" can even be considered wrong, from the readers perspective: Both previous tests do not have an editor. Just an empty vector. There are no type annotations, so you don't even know what the types inside the vector are.
Please, highlight the fact, that unlike the other two tests, this actual uses another editor.

Reply:
It's not possible to use `sort_by` with something else than editors. There is no other way to use `sort_by`. This is a minimal example that `sort_by` works. Whatever the minimal example showing that `sort_by` works is.

Reply:
To add to my previous answer: I am not the only one writing tests here. The Rust type system and serde are ensuring these things here too. Writing explicit code that the compiler verifies can be better than writing tests for everything. And this is pretty explicitly a `Vec<Editor>`.